File: frontend/types/sentiment.ts
export interface SentimentRequest {
  text: string
  language?: string
  include_emotions?: boolean
  include_toxicity?: boolean
  include_entities?: boolean
}

export interface SentimentResponse {
  sentiment: 'positive' | 'negative' | 'neutral'
  confidence: number
  probabilities: {
    positive: number
    negative: number
    neutral: number
  }
  emotions?: Record<string, number>
  toxicity?: {
    is_toxic: boolean
    confidence: number
  }
  entities?: Array<{
    text: string
    label: string
  }>
  language: string
  processing_time_ms: number
  timestamp: string
}

export interface BatchSentimentRequest {
  texts: string[]
  language?: string
  include_emotions?: boolean
}

export interface BatchSentimentResponse {
  results: SentimentResponse[]
  batch_id: string
  total_processing_time_ms: number
  successful_count: number
  failed_count: number
}

File: backend/app/schemas/sentiment.py
from pydantic import BaseModel, Field, validator
from typing import Optional, Dict, List
from datetime import datetime

class SentimentRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000)
    language: Optional[str] = "auto"
    include_emotions: bool = True
    include_toxicity: bool = False
    include_entities: bool = False
    
    @validator('text')
    def validate_text(cls, v):
        if not v.strip():
            raise ValueError('Text cannot be empty or only whitespace')
        return v.strip()

class SentimentResponse(BaseModel):
    sentiment: str  # positive, negative, neutral
    confidence: float
    probabilities: Dict[str, float]
    emotions: Optional[Dict[str, float]] = None
    toxicity: Optional[Dict[str, float]] = None
    entities: Optional[List[Dict[str, str]]] = None
    language: str
    processing_time_ms: float
    timestamp: datetime = Field(default_factory=datetime.utcnow)

class BatchSentimentRequest(BaseModel):
    texts: List[str] = Field(..., max_items=100)
    language: Optional[str] = "auto"
    include_emotions: bool = True
    include_toxicity: bool = False

class BatchSentimentResponse(BaseModel):
    results: List[SentimentResponse]
    batch_id: str
    total_processing_time_ms: float
    successful_count: int
    failed_count: int